Mathlib/Order/SymmDiff.lean

```diff
@@ -0,0 +1,872 @@
+/-
+Copyright (c) 2021 Bryan Gin-ge Chen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz, Bryan Gin-ge Chen, Yaël Dillies
+Ported by: Frédéric Dupuis
+-/
+import Mathlib.Order.BooleanAlgebra
+import Mathlib.Logic.Equiv.Basic
+
+/-!
+# Symmetric difference and bi-implication
+
+This file defines the symmetric difference and bi-implication operators in (co-)Heyting algebras.
+
+## Examples
+
+Some examples are
+* The symmetric difference of two sets is the set of elements that are in either but not both.
+* The symmetric difference on propositions is `xor`.
+* The symmetric difference on `Bool` is `bxor`.
+* The equivalence of propositions. Two propositions are equivalent if they imply each other.
+* The symmetric difference translates to addition when considering a Boolean algebra as a Boolean
+  ring.
+
+## Main declarations
+
+* `symmDiff`: The symmetric difference operator, defined as `(a \ b) ⊔ (b \ a)`
+* `bihimp`: The bi-implication operator, defined as `(b ⇨ a) ⊓ (a ⇨ b)`
+
+In generalized Boolean algebras, the symmetric difference operator is:
+
+* `symmDiff_comm`: commutative, and
+* `symmDiff_assoc`: associative.
+
+## Notations
+
+* `a ∆ b`: `symmDiff a b`
+* `a ⇔ b`: `bihimp a b`
+
+## References
+
+The proof of associativity follows the note "Associativity of the Symmetric Difference of Sets: A
+Proof from the Book" by John McCuan:
+
+* <https://people.math.gatech.edu/~mccuan/courses/4317/symmetricdifference.pdf>
+
+## Tags
+
+boolean ring, generalized boolean algebra, boolean algebra, symmetric difference, bi-implication,
+Heyting
+-/
+
+
+open Function OrderDual
+
+variable {ι α β : Type _} {π : ι → Type _}
+
+/-- The symmetric difference operator on a type with `⊔` and `\` is `(A \ B) ⊔ (B \ A)`. -/
+def symmDiff [HasSup α] [SDiff α] (a b : α) : α :=
+  a \ b ⊔ b \ a
+#align symm_diff symmDiff
+
+/-- The Heyting bi-implication is `(b ⇨ a) ⊓ (a ⇨ b)`. This generalizes equivalence of
+propositions. -/
+def bihimp [HasInf α] [HImp α] (a b : α) : α :=
+  (b ⇨ a) ⊓ (a ⇨ b)
+#align bihimp bihimp
+
+/- This notation might conflict with the Laplacian once we have it. Feel free to put it in locale
+  `order` or `symm_diff` if that happens. -/
+/-- Notation for symmDiff -/
+infixl:100 " ∆ " =>  symmDiff
+
+/-- Notation for bihimp -/
+infixl:100 " ⇔ " => bihimp
+
+theorem symmDiff_def [HasSup α] [SDiff α] (a b : α) : a ∆ b = a \ b ⊔ b \ a :=
+  rfl
+#align symm_diff_def symmDiff_def
+
+theorem bihimp_def [HasInf α] [HImp α] (a b : α) : a ⇔ b = (b ⇨ a) ⊓ (a ⇨ b) :=
+  rfl
+#align bihimp_def bihimp_def
+
+theorem symmDiff_eq_xor (p q : Prop) : p ∆ q = Xor' p q :=
+  rfl
+#align symm_diff_eq_xor symmDiff_eq_xor
+
+@[simp]
+theorem bihimp_iff_iff {p q : Prop} : p ⇔ q ↔ (p ↔ q) :=
+  (iff_iff_implies_and_implies _ _).symm.trans Iff.comm
+#align bihimp_iff_iff bihimp_iff_iff
+
+@[simp]
+theorem Bool.symmDiff_eq_bxor : ∀ p q : Bool, p ∆ q = xor p q := by decide
```

Comment:
There should be no mention of `bxor` in this file -- it no longer exists. This theorem should be called `symmDiff_eq_xor`.